lib/MySQL_Session.cpp

```diff
@@ -5075,6 +5078,12 @@ int MySQL_Session::handler() {
 							handler_minus1_HandleBackendConnection(myds, myconn);
 						}
 					} else {
+						if (active_transactions == 0) {
```

Comment:
I think this is the beginning of a query. When `rc==0` is the end of the query and when `rc==-1` the query do not begin because it has failed.

What I am realizing now is that if `rc==-1` because proxysql failed to retrieve the result of a query, then the number of active transaction is not going to be updated and the `transaction_started_at` is not going to be reset.